src/jit/hwintrinsicxarch.cpp

```diff
@@ -337,11 +508,60 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return gtNewIconNode(issupported);
     }
-    else if (!issupported)
+    // - calling to unsupported intrinsics must throw PlatforNotSupportedException
+    // - 64-bit (r64) intrinsics must throw PlatforNotSupportedException on 32-bit platform
+    else if (!issupported || is64bitOnlyIntrinsicOn32bitPlatform(category, callType))
     {
         return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
     }
 
+    // table-driven importer of simple intrinsics
+    if (impIsTableDrivenHWIntrinsic(category))
+    {
+        unsigned int sizeBytes;
+        var_types    baseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &sizeBytes);
+        assert(baseType != TYP_UNKNOWN && sizeBytes != 0);
+        var_types               retType  = getSIMDTypeForSize(sizeBytes);
+        unsigned                simdSize = simdSizeOfHWIntrinsic(intrinsic, sig);
+        CORINFO_ARG_LIST_HANDLE argList  = sig->args;
+        CORINFO_CLASS_HANDLE    argClass;
+
+        assert(numArgs > 0);
+        assert(retType != TYP_UNDEF);
+        assert(retType == TYP_SIMD16 || retType == TYP_SIMD32);
+        assert(insOfHWIntrinsic(intrinsic, baseType) != INS_invalid);
+        assert(simdSize == 32 || simdSize == 16);
+
+        GenTree*  retNode = nullptr;
+        GenTree*  op1     = nullptr;
+        GenTree*  op2     = nullptr;
+        var_types op1Type = getArgTypeFromSignature(argList, sig);
+        var_types op2Type = TYP_UNKNOWN;
+
+        switch (numArgs)
+        {
+            case 1:
+                op1     = impSIMDPopStack(op1Type);
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, baseType, simdSize);
+                break;
+            case 2:
+                argList = info.compCompHnd->getArgNext(argList);
+                op2Type = getArgTypeFromSignature(argList, sig);
+                op2     = impSIMDPopStack(op2Type);
```

Comment:
I don't think we can use `impSIMDPopStack` for non-SIMD types. IIRC, it causes issues with non-SIMD types due to other checks/handling it does.

Reply:
Thank you! Will change.